git_revision.cpp.in

```diff
@@ -1,10 +1,33 @@
 // This file is used by CMake to generate `$CMAKE_CURRENT_BINARY_DIR/git_revision.cpp`,
 // which is used by CCI to make up the compiler's build version.
 
-auto cci_git_refspec() -> const char *;
-auto cci_git_hash() -> const char *;
-auto cci_git_tag() -> const char *;
+#cmakedefine CCI_GIT_REFSPEC "@GIT_REFSPEC@"
+#cmakedefine CCI_GIT_HASH "@GIT_HASH@"
+#cmakedefine CCI_GIT_TAG "@GIT_TAG@"
```

Comment:
Doesn't `#cmakedefine` require that those `CCI_GIT_*` are set in CMake? Right now the behavior of this commit is to always produce `/* #undef CCI_GIT_* */`.

Reply:
Oops, you are totally right.
I meant to use a simple `#define` here, not `#cmakedefine`.